vaadin-radio-button-flow-demo/src/main/java/com/vaadin/flow/component/radiobutton/demo/RadioButtonGroupView.java

```diff
@@ -15,330 +15,287 @@
     }
 
-    private void addComponentWithThemeVariant() {
+    private void basicDemo() {
         // begin-source-example
-        // source-example-heading: Theme variants usage
-        RadioButtonGroup<String> group = new RadioButtonGroup<>();
-        group.setItems("foo", "bar", "baz");
-        group.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+        // source-example-heading: Basic usage
+        RadioButtonGroup<String> radioGroup = new RadioButtonGroup<>();
+        radioGroup.setLabel("Label");
+        radioGroup.setItems("Option one", "Option two", "Option three");
+        radioGroup.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+        radioGroup.setValue("Option one");
         // end-source-example
 
-        addVariantsDemo(() -> {
-            return group;
-        }, GeneratedVaadinRadioGroup::addThemeVariants,
-                GeneratedVaadinRadioGroup::removeThemeVariants,
-                RadioGroupVariant::getVariantName,
-                RadioGroupVariant.LUMO_VERTICAL);
+        addCard("Basic usage", radioGroup);
     }
 
-    private void addBasicFeatures() {
-        Div message = new Div();
-
+    private void disabledAndDisabledItem() {
         // begin-source-example
-        // source-example-heading: Basic radio button group
-        RadioButtonGroup<String> group = new RadioButtonGroup<>();
-        group.setItems("foo", "bar", "baz");
-        group.addValueChangeListener(event -> message.setText(String.format(
-                "Radio button group value changed from '%s' to '%s'",
-                event.getOldValue(), event.getValue())));
+        // source-example-heading: Disabled state
+        RadioButtonGroup<String> disabledRadioGroup = new RadioButtonGroup<>();
+        disabledRadioGroup.setLabel("Disabled");
+        disabledRadioGroup.setItems("Option one", "Option two", "Option three");
+        disabledRadioGroup.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+        disabledRadioGroup.setValue("Option one");
+        disabledRadioGroup.setEnabled(false);
+
+        RadioButtonGroup<String> radioGroup = new RadioButtonGroup<>();
+        radioGroup.setLabel("Disabled item");
+        radioGroup.setItems("Option one", "Option two", "Option three");
+        radioGroup.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+        radioGroup.setItemEnabledProvider(item -> !"Option three".equals(item));
         // end-source-example
 
-        group.setId("button-group-with-value-change-listener");
-        message.setId("button-group-value");
-
-        addCard("Basic radio button group", group, message);
+        disabledRadioGroup.getStyle().set("margin-right", "7.5em");
+        VerticalLayout verticalLayout = new VerticalLayout(disabledRadioGroup,
+                radioGroup);
+        addCard("Disabled state", verticalLayout);
     }
 
-    private void addItemRenderer() {
-        Div message = new Div();
-
+    private void readonlyRadioButton() {
```

Comment:
As the component doesn't have a proper readonly mode yet, we could hide this demo for now